### ISSUE TITLE
TypeError for tasks.cfg

### ISSUE THREAD

**Issue by HannoSpreeuw** [tikk3r/lofar-grid-hpccloud#8]
For some reason prefactor does not give the error that the Ubuntu-Kern Singularity container - launched through
`singularity exec ubuntu_kern.simg genericpipeline.py -d -c pipeline.cfg Pre-Facet-Calibrator.parset`
- gives for the same pipeline configuration and calibrator parset:

> TypeError: ['/cm/shared/package/lofar/2.21.4-gcc-4.9.3/share/pipeline/tasks.cfg'] is an invalid value for FileList task_files
No results

Any idea why?

**Comment by tikk3r** [comment]
This is probably due to a wrong values in the `[DEFAULT]` section of `pipeline.cfg`. This is what I use for now:

```
[DEFAULT]
lofarroot=/usr/
casaroot=/usr/
pyraproot= 
hdf5root=/opt/hdf5
wcsroot=/usr/lib
aoflaggerroot=
pythonpath=/usr/lib/python2.7/dist-packages/
runtime_directory=rundir
recipe_directories=[%(pythonpath)s/lofar,%(pythonpath)s/lofarpipe/recipes,/path/to/prefactor/]
working_directory=workdir
task_files = [/usr/share/pipeline/tasks.cfg]
```
Guidelines for this will be added once the images are stable enough.

**Comment by tikk3r** [comment]
That's only partially true. The cluster configuration part in the `pipeline.cfg` will indeed be different for you and me. The paths in the DEFAULT section, however, pointing to the various software locations will not be different, as that all resides in the container.

The error occurs because the pipeline cannot find `tasks.cfg`, which is likely because your `task_files` does not point to the value in my above snippet (I've added the prefactor paths to the recipe_directories there as well, they were missing).

**Comment by HannoSpreeuw** [comment]
Ah. Thanks a lot for that explanation. That enabled some progress. I did have to replace `rundir` by an actual path, because `rundir` is not defined as far as I can see.
Now I am running into a `saved state` problem:

> PipelineException: Input task_files (['/usr/share/pipeline/tasks.cfg']) differs from saved state (['/cm/shared/package/lofar/2.21.4-gcc-4.9.3/share/pipeline/tasks.cfg'])

How can I fix that?

**Comment by tikk3r** [comment]
You are right, rundir and workdir do have to be adjusted to where you work. To fix that, there are two options:

1 Delete and start again fresh.
2 Edit the `statefile` (it's just a text file) in the rundir directory and replace `/cm/shared/package/lofar/2.21.4-gcc-4.9.3/share/pipeline/tasks.cfg` with `/usr/share/pipeline/tasks.cfg`.

**Comment by HannoSpreeuw** [comment]
Thanks, that helped.
I am now running into

> 2018-11-21 17:59:59 ERROR   genericpipeline: [Errno 2] No such file or directory: '/var/scratch/hspreeuw/data/'

That must be referring to a line in my `Pre-Facet-Calibrator.parset`:
`! cal_input_path           =  /var/scratch/hspreeuw/data/`

That directory exists and contains a Measurement Set, so I don't understand this error.

**Comment by tikk3r** [comment]
Try binding the path with `singularity exec -B /var/scratch/hspreeuw/data/ ubuntu_kern.simg genericpipeline.py -d -c pipeline.cfg Pre-Facet-Calibrator.parset`. By default I think only the current directory and a few system directories are available.

**Comment by HannoSpreeuw** [comment]
Thanks, that worked. Now running into this:

> 2018-11-21 21:14:54 DEBUG   genericpipeline.executable_args: compute.dispatch results job 0: job_duration: 0.719278097153, returncode: 1 
2018-11-21 21:14:55 DEBUG   genericpipeline.executable_args: Adding node_logging_information
2018-11-21 21:14:55 ERROR   genericpipeline.executable_args: A job has failed with returncode 1 and error_tolerance is not set. Bailing out!
2018-11-21 21:14:55 WARNING genericpipeline.executable_args: Note: recipe outputs are not complete
2018-11-21 21:14:55 WARNING genericpipeline.executable_args: recipe executable_args completed with errors
2018-11-21 21:14:55 WARNING genericpipeline: pythonplugin reports failure (using executable_args recipe)
2018-11-21 21:14:55 ERROR   genericpipeline: *******************************************
2018-11-21 21:14:55 ERROR   genericpipeline: Failed pipeline run: Pre-Facet-Calibrator
2018-11-21 21:14:55 ERROR   genericpipeline: Detailed exception information:
2018-11-21 21:14:55 ERROR   genericpipeline: <class 'lofarpipe.support.lofarexceptions.PipelineRecipeFailed'>
2018-11-21 21:14:55 ERROR   genericpipeline: pythonplugin failed
2018-11-21 21:14:55 ERROR   genericpipeline: *******************************************
2018-11-21 21:14:55 ERROR   genericpipeline: LOFAR Pipeline finished unsuccesfully.
2018-11-21 21:14:55 WARNING genericpipeline: recipe genericpipeline completed with errors
Results:
